# NB: THIS IS A DUMMY SCRIPT, only run using snakemake -n

input_data = "input_data/{id}.fastq"
ids, = glob_wildcards(input_data)
subsampling = [1000, 10000]

# use a config file to store our metadata
configfile: "config.yaml"

rule dummy_mapping:
    input:
        fastq=input_data,
        index=lambda wc: config["sample_index"][wc.id]
    output: "output/{id}_subsample-{subsample}.bam"
    params:
        subsample="--subsample {subsample}"
    shell:
        "dummy_mapper -f {input.fastq} "
        "-i {input.index} "
        "-o {output} "
        "{params.subsample} "

# in reality, we would use something like
# STAR --genomeDir {input.index} --readFilesIn {input.fastq}
# --outFileNamePrefix output/{wildcards.id}

rule all:
    input: expand(rules.dummy_mapping.output, id=ids, subsample=subsampling)
